uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/absence-tab/components/absence-tab-work-patterns_test.js

```diff
@@ -24,7 +44,73 @@
         expect($log.debug).toHaveBeenCalled();
       });
 
-      function compileComponent() {
+      describe('init()', function () {
+        describe('when custom Work patterns are present', function () {
+          it('Loads custom Work patterns', function () {
+            expect(controller.customWorkpattern.length).toEqual(workPatternData.workPatternsOf.values.length);
+          });
+
+          it('Assign correct change reason label', function () {
+            _.each(controller.customWorkpattern, function (customWorkpattern) {
+              var changeReasonLabel = optionGroupMock.getCollection('hrjc_revision_change_reason').find(function (reason) {
+                return customWorkpattern.change_reason === reason.value;
+              }).label;
+
+              expect(customWorkpattern.change_reason_label).toBe(changeReasonLabel);
+            });
+          });
+        });
+
+        describe('when custom Work patterns are not present', function () {
+          var defaultWorkPattern = workPatternData.getAllWorkPattern.values[0];
+
+          beforeEach(function () {
+            spyOn(WorkPattern, 'workPatternsOf').and.returnValue($q.resolve([]));
+            spyOn(WorkPattern, 'default').and.returnValue($q.resolve(defaultWorkPattern));
+            compileComponent();
+          });
+
+          it('Loads default Work pattern', function () {
+            expect(controller.defaultWorkPattern).toEqual(defaultWorkPattern);
+          });
+        });
+      });
+
+      describe('deleteWorkPattern()', function () {
+        var confirmFunction;
+        var contactWorkPatternID = '2';
+
+        beforeEach(function () {
+          spyOn(dialog, 'open').and.callFake(function (params) {
+            confirmFunction = params.onConfirm;
+          });
+          controller.deleteWorkPattern(contactWorkPatternID);
+        });
+
+        it('opens dialog box to confirm deletion', function () {
+          expect(dialog.open).toHaveBeenCalledWith(jasmine.objectContaining({
+            title: 'Confirm Cancellation?',
+            copyCancel: 'Cancel',
+            copyConfirm: 'Confirm',
+            classConfirm: 'btn-danger',
+            msg: 'This cannot be undone',
+            onConfirm: jasmine.any(Function)
+          }));
+        });
+
+        describe('when deletion is confirmed', function () {
+          beforeEach(function () {
+            spyOn(WorkPattern, 'unassignWorkPattern').and.returnValue($q.resolve([]));
+            confirmFunction();
+          });
+
+          it('Contact work pattern is un assigned', function () {
+            expect(WorkPattern.unassignWorkPattern).toHaveBeenCalledWith(contactWorkPatternID);
+          });
+        });
+      });
+
+      function compileComponent () {
         var $scope = $rootScope.$new();
         var contactId = CRM.vars.leaveAndAbsences.contactId;
```

Comment:
Yes the values are already set here 
https://github.com/civicrm/civihr/blob/5cae3adac4f78ef8a209a687a89790541fb1ae4a/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/globals.js
to be used in test cases, so this should be fine.